### PR TITLE
Make nightwatch err and warn visible in stdout

### PR DESCRIPTION
**Pull request by g00dnatur3**
Currently the "ERROR" and "WARN" logs from nightwatch are suppressed when the `debug` flag is turned OFF..
and our customers ALWAYS use the `debug` off, because if you turn it on yourlog will be filled with base64 screenshot gobbledegook...

Also, the approach i took, we will see all the "ERROR-ed" selenium request/response logs in our magellan log

BEFORE LOG

Screen Shot 2020-08-24 at 3 42 39 PM

AFTER LOG

Screen Shot 2020-08-24 at 3 37 24 PM

in the example above i modified the nightwatch code in my node_modules to have this code `Logger.warn('__EXAMPLE_WORKING__ Success saving screenshot 200 ok');`

Previously, this code would not have been logged to the magellan stdout.

in the AFTER LOG you can see that it is working GREAT :)

the `â` is really the red X .. its just my terminal settings are making it show up as `â` instead of the fancy `X`

### CODE REVIEW

File: src/cli.js [hunk 1]
```diff
@@ -104,6 +104,11 @@ module.exports = {
                 rootWorkingDirectory: process.cwd()
               });
         }
+        if (settings.framework === 'testarmada-magellan-nightwatch-plugin') {
```
Comment (zzo) [inline src/cli.js:107]: I thought we were just gonna pass this in as a command-line arg?  This is sorta ugly

File: src/util/childProcess.js [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
+// relevant code for this feature:
+//   STDOUT_WHITE_LIST, SLICE_ON_TEXT, infoSlicer, isTextWhiteListed
+// ------------------------------------------------------------------------------------------------
+
+// if the "this.handler.stdout" stream of the childprocess does not 
+// include atleast one of these tokens then it will not be included in the "this.stdout"
+const STDOUT_WHITE_LIST = ['ERROR', 'WARN', 'Test Suite', '✖']
+
+// we slice the VERBOSE nighwatch stdout stream on the purple INFO text that has black background
+const SLICE_ON_TEXT = '\033[1;35m\033[40mINFO\033[0m'
```
Comment (zzo) [inline src/util/childProcess.js:32]: do you know if this works for both nightwatch .9 and 1.x?

File: src/util/childProcess.js [hunk 3]
```diff
@@ -48,21 +74,24 @@ module.exports = class ChildProcess {
     }
   }
 
+  isTextWhiteListed(text) {
```
Comment (zzo) [inline src/util/childProcess.js:77]: return STDOUT_WHITE_LIST.some(item => text.includes(item))
```suggestion
  isTextWhiteListed(text) {
    return STDOUT_WHITE_LIST.some(item => text.includes(item))
  }
}
```
?

File: src/util/childProcess.js (hunk 2 above)
Comment (g00dnatur3) [reply src/util/childProcess.js:32]: good question.

looking at their logger in the master:
https://github.com/nightwatchjs/nightwatch/blob/master/lib/utils/logger.js

it seems it should work just fine cause they havent changed the logger INFO coloring at all

Comment (g00dnatur3) [reply src/util/childProcess.js:32]: https://github.com/nightwatchjs/nightwatch/blob/a65f49e5cc3f6ac3c23eac8c9cbb3da35001c93a/lib/utils/logger.js#L142

File: src/cli.js (hunk 1 above)
Comment (g00dnatur3) [reply src/cli.js:107]: the whole idea it to make this the "new" default..

we are not being fully verbose in our magellan log, in the child process we are filtering out the "verbose" stuff and only shinning a light on the errors and warns

so this should be the default, its the whole point...

opts.argv.debug = true. just tells niughtwatch to be verbose...


i guess we can say if the (opts.argv.debug) is already true then we dont filter out anything in the child process...

you know what im saying?

**Pull request merged**